src/main/java/org/folio/list/services/CustomTenantService.java

```diff
@@ -1,37 +1,138 @@
 public class CustomTenantService extends TenantService {
 
   protected final PrepareSystemUserService prepareSystemUserService;
+  private final ListRepository listRepository;
+  @Autowired
+  private ObjectMapper objectMapper;
 
   @Autowired
   public CustomTenantService(
     JdbcTemplate jdbcTemplate,
     FolioExecutionContext context,
     FolioSpringLiquibase folioSpringLiquibase,
-    PrepareSystemUserService prepareSystemUserService
+    PrepareSystemUserService prepareSystemUserService,
+    ListRepository listRepository
   ) {
     super(jdbcTemplate, context, folioSpringLiquibase);
     this.prepareSystemUserService = prepareSystemUserService;
+    this.listRepository = listRepository;
   }
 
   @Override
   protected void afterTenantUpdate(TenantAttributes tenantAttributes) {
     log.info("Initializing system user");
-    prepareSystemUserService.setupSystemUser();
+   prepareSystemUserService.setupSystemUser();
+    Collection<ListEntity> lists = (Collection<ListEntity>) listRepository.findAll();
+    for (ListEntity list : lists) {
+      System.out.println(list.getEntityTypeId());
+      if (list.getEntityTypeId().equals(UUID.fromString("0069cf6f-2833-46db-8a51-8934769b8289"))) {
+        list.setEntityTypeId(UUID.fromString("ddc93926-d15a-4a45-9d9c-93eadc3d9bbf"));
+        try {
+          updateFqlQueryForUsers(list);
+          updateFieldNamesForUsers(list);
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+      listRepository.save(list);
+    }
+  }
+
+  public void updateFqlQueryForUsers(ListEntity list) throws Exception {
+    Map<String, String> keyMappings = getStringStringMap();
+    String currentQuery = list.getFqlQuery();
+    Map<String, Object> map = objectMapper.readValue(currentQuery, new TypeReference<HashMap<String, Object>>() {});
+
+    Map<String, Object> result = new HashMap<>();
+
+    for (Map.Entry<String, Object> entry : map.entrySet()) {
+      String key = entry.getKey();
+      System.out.println("print key" +key);
+      if (keyMappings.containsKey(key)) {
+        String newKey = keyMappings.get(key);
+        result.put(newKey, entry.getValue());
+      } else {
+        result.put(key, entry.getValue());
+      }
+    }
+    String resultStr = objectMapper.writeValueAsString(result);
+    System.out.print("new migrated query" +resultStr);
+    list.setFqlQuery(resultStr);
+  }
+
+
+  public void updateFieldNamesForUsers(ListEntity list){
+    Map<String, String> keyMappings = getStringStringMap();
+    List<String> listFields = list.getFields();
+    List<String> updatedFields = new ArrayList<>();
+    for (String field : listFields) {
+      // If no mapping is found, retain the original field name
+      updatedFields.add(keyMappings.getOrDefault(field, field));
+    }
+    list.setFields(updatedFields); // Update the fields in ListEntity with the translated names
+
+  }
+
+  @NotNull
+  private static Map<String, String> getStringStringMap() {
```

Comment:
Since this is going to keep getting bigger and bigger, I wonder if it'd be better to dump all of it into some JSON5 files or something. That way, the code doesn't get cluttered up with huge maps

Reply:
either that or we make a class to hold these constant maps